web/modules/custom/dpl_event/src/EventWrapper.php

```diff
@@ -122,7 +124,18 @@ public function getState(): ?EventState {
     }
 
     $states = array_map(function (array $value) {
-      return EventState::from($value['value']);
+      $value = $value['value'] ?? NULL;
+      $enum = EventState::tryFrom($value);
+
+      if (is_null($enum)) {
+        $logger = DrupalTyped::service(LoggerInterface::class, 'dpl_event.logger');
+
+        $logger->error("Could not map '@value' to EventState enum.", [
+          '@value' => $value ?? 'NULL',
+        ]);
+      }
+
+      return $enum;
```

Comment:
```suggestion
      try {
        return EventState::from($value['value']);
      } catch (\ValueError $e) {
         $this->logger->error($e->getMessage())
         return NULL;
      }
```

I think this can be simplified a bit using the exceptions already provided.

Note that the logger should be injected in the constructor instead of using `DrupalTyped`.

Reply:
In general I am a bit on the fence with this change.

As I see it we should check that our data is valid before we store it. Thus we could assume that the value will always be valid. On the other hand I see that there might be situations where that is not the case - perhaps we remove a value from an enum or the like. With that in mind I think this is OK.